### ISSUE TITLE
Support c/c++ reserved keywords for field names when using '-nif'

### ISSUE THREAD

**Issue by jabarszcz** [tomas-abrahamsson/gpb#146]
`protoc` seems to add an underscore suffix to field names when they are c/c++ reserved words :  
```
$ cat > x.proto <<EOF                                                                                                                                                
syntax="proto2";
message a_message {
    optional bool private = 1;
}
EOF

$ ./gpb/bin/protoc-erl -I. x.proto -nif
$ protoc --cpp_out=$PWD x.proto
$ g++ -fPIC -shared -O3 -o x.nif.so x.nif.cc x.pb.cc $(pkg-config --cflags protobuf) $(pkg-config --libs protobuf) -I /usr/lib/erlang/erts-9.3.3/include/
x.nif.cc: In function ‘int p_msg_a_message(ErlNifEnv*, ERL_NIF_TERM, a_message*)’:
x.nif.cc:86:16: error: ‘class a_message’ has no member named ‘set_private’; did you mean ‘set_private_’?
             m->set_private(1);
                ^~~~~~~~~~~
                set_private_
x.nif.cc:88:16: error: ‘class a_message’ has no member named ‘set_private’; did you mean ‘set_private_’?
             m->set_private(1);
                ^~~~~~~~~~~
                set_private_
x.nif.cc:90:16: error: ‘class a_message’ has no member named ‘set_private’; did you mean ‘set_private_’?
             m->set_private(0);
                ^~~~~~~~~~~
                set_private_
x.nif.cc: In function ‘ERL_NIF_TERM u_msg_a_message(ErlNifEnv*, const a_message*)’:
x.nif.cc:139:13: error: ‘const class a_message’ has no member named ‘has_private’; did you mean ‘has_private_’?
     if (!m->has_private())
             ^~~~~~~~~~~
             has_private_
x.nif.cc:142:16: error: expected unqualified-id before ‘private’
         if (m->private())
                ^~~~~~~
x.nif.cc:142:16: error: expected ‘)’ before ‘private’
         if (m->private())
            ~   ^~~~~~~
                )
```
The list of keywords for protoc : https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/compiler/cpp/cpp_helpers.cc#L75-L88
The logic for field/enum value names : https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/compiler/cpp/cpp_helpers.cc#L294-L309

**Comment by tomas-abrahamsson** [comment]
Thanks for the links to how it is handled. I've pushed a fix to a branch `c++-keywords` in case you'd like to take a look or test if it solves your issue.

**Comment by jabarszcz** [comment]
I have added comments about a small issue with capitalization. Otherwise it seems good to me!

**Comment by tomas-abrahamsson** [comment]
I just (force-)pushed an update that fixes the type spec issue as well as the capitalization issue. I'll go ahead and make a new release for both issues shortly

**Comment by jabarszcz** [comment]
Great! Thanks! :100:

**Comment by tomas-abrahamsson** [comment]
Included in just-pushed 4.2.2, closing.
